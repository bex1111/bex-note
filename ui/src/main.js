import { createApp } from 'vue'
import App from './App.vue'
import PrimeVue from 'primevue/config';
import Button from 'primevue/button';
import Select from 'primevue/select';
import Toolbar from 'primevue/toolbar';
import Card from 'primevue/card';
import InputText from 'primevue/inputtext';
import FloatLabel from 'primevue/floatlabel';
import Fluid from 'primevue/fluid';
import Dialog from 'primevue/dialog';
import Aura from '@primeuix/themes/aura';
import 'primeicons/primeicons.css'
import { createPinia } from 'pinia'
import {useAuthorizationStore} from "./store/authorization";
import './style.css'

const app=createApp(App)

const pinia = createPinia()
app.use(pinia)
export const tokenStore = useAuthorizationStore()

app.use(PrimeVue,{
    theme: {
        preset: Aura
    }
});
app.component('prime-button', Button);
app.component('prime-select', Select);
app.component('prime-toolbar', Toolbar);
app.component('prime-card', Card);
app.component('prime-input', InputText);
app.component('prime-float-label', FloatLabel);
app.component('prime-fluid', Fluid);
app.component('prime-dialog', Dialog);

app.mount('#app')
