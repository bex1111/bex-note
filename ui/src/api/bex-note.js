import axios from 'axios';
import {tokenStore} from '../main';


export const getNoteList = async () => {
    const response = await axios.get('http://localhost:1080/api/internal/note/list', {
        headers: {
            'x-auth-token': tokenStore.token
        }
    });
    return response.data;
}

export const authorize = async (username, password) => {
    const response = await axios.post('http://localhost:1080/api/authorization', {
        username,
        password
    });
    return response.data;
}


export const deleteNote = async (title) => {
    const response = await axios.delete('http://localhost:1080/api/internal/note/delete', {
        headers: {
            'x-auth-token': tokenStore.token
        },
        data: { title }
    });
    return response.data;
}

export const saveNote = async (title,content) => {
    const response = await axios.delete('http://localhost:1080/api/internal/note/save', {
        headers: {
            'x-auth-token': tokenStore.token
        },
        data: { title, content }
    });
    return response.data;
}


