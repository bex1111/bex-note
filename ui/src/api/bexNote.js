import axios from 'axios';
import {tokenStore} from '../main';


export const getNoteList = async () => {
    const response = await axios.get('/api/internal/note/list', {
        headers: {
            'x-auth-token': tokenStore.token
        }
    });
    return response.data;
}

export const authorize = async (username, password) => {
    const response = await axios.post('/api/authorize', {
        username,
        password
    });
    return response.data;
}


export const deleteNote = async (title) => {
    const response = await axios.delete('/api/internal/note/delete', {
        headers: {
            'x-auth-token': tokenStore.token
        },
        data: {title}
    });
    return response.data;
}

export const saveNote = async (title, content) => {
    const response = await axios.post('/api/internal/note/save', {
        data: {title, content}
    }, {
        headers: {
            'x-auth-token': tokenStore.token
        }
    });
    return response.data;
}

export const getContent = async (title) => {
    const response = await axios.post('/api/internal/note/content', {
            data: {title}
        },
        {
            headers: {
                'x-auth-token': tokenStore.token
            }
        });
    return response.data;
}
