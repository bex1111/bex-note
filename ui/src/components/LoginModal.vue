<script setup>

import {ref} from "vue";
import {tokenStore} from "../main";
import {authorize} from "../api/bexNote";

const visible = ref(true);
const username = ref();
const password = ref();

const closeCallback = async () => {
  tokenStore.$patch({
    token: await authorize(username.value, password.value)
  });
  visible.value = false;
}
</script>

<template>
  <prime-dialog v-model:visible="visible" pt:root:class="!border-0 !bg-transparent" pt:mask:class="backdrop-blur-sm"
                :closable="false">
    <div class="flex flex-col px-8 py-8 gap-6 rounded-2xl">
      <div class="inline-flex flex-col gap-2">
        <label for="username" class="text-primary-50 font-semibold">Username</label>
        <prime-input id="username" v-model="username"></prime-input>
      </div>
      <div class="inline-flex flex-col gap-2">
        <label for="password" class="text-primary-50 font-semibold">Password</label>
        <prime-input id="password"
                     v-model="password"
                     type="password"></prime-input>
      </div>
      <div class="flex items-center gap-4">
        <prime-button label="Sign-In" @click="closeCallback" variant="text"
                      class="!p-4 w-full !text-primary-50 !border !border-white/30 hover:!bg-white/10"></prime-button>
      </div>
    </div>
  </prime-dialog>
</template>

<style scoped>

</style>