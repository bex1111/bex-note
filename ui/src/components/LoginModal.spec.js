import {mount} from '@vue/test-utils';
import {vi} from 'vitest';
import LoginModal from './LoginModal.vue';
import {authorize} from '../api/bexNote';
import {tokenStore} from '../main';

vi.mock('../api/bexNote', () => ({
    authorize: vi.fn().mockResolvedValue({token: 'abc'}),
}));

const mockSubscriptionCallback = vi.fn();
vi.mock('../main', () => ({
    tokenStore: {
        token: null,
        $patch: vi.fn(),
        $subscribe: vi.fn((callback) =>
            mockSubscriptionCallback.mockImplementation(callback)
        ),
        saveToken: vi.fn()
    }
}));

describe('LoginModal.vue', () => {

    const createWrapper = () => mount(LoginModal, {
        global: {
            stubs: {
                'prime-dialog': {
                    template: '<div class="prime-dialog-stub"><slot></slot></div>'
                },
                'prime-input': true,
                'prime-button': true
            }
        }
    })

    it('matches snapshot', async () => {
        const wrapper = await createWrapper();
        expect(wrapper.html()).toMatchSnapshot();
    });

    describe('closeCallback', () => {
        it('emits login event and hides modal when called', async () => {
            const wrapper = await createWrapper();

            wrapper.vm.username = 'user';
            wrapper.vm.password = 'pass';
            await wrapper.vm.closeCallback();

            expect(wrapper.emitted('login')).toBeTruthy();
            expect(wrapper.emitted('login')).toHaveLength(1);
            expect(wrapper.vm.visible).toBe(false);
            expect(authorize).toHaveBeenCalledWith(wrapper.vm.username, wrapper.vm.password);
            expect(tokenStore.saveToken).toHaveBeenCalledWith('abc');
        });
    });

    describe('tokenStore subscription', () => {
        test.each([
            {
                description: 'shows modal when token becomes null',
                tokenValue: null,
                expectedVisible: true
            },
            {
                description: 'does not show modal when token has value',
                tokenValue: 'some-token',
                expectedVisible: false
            }
        ])('$description', ({tokenValue, expectedVisible}) => {
            const wrapper = createWrapper();
            tokenStore.token = tokenValue;

            mockSubscriptionCallback();

            expect(wrapper.vm.visible).toBe(expectedVisible);
        });
    });
});
