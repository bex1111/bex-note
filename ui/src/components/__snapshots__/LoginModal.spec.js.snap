// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginModal > matches snapshot 1`] = `
"<div class=\\"prime-dialog-stub\\" visible=\\"true\\" pt:root:class=\\"!border-0 !bg-transparent\\" pt:mask:class=\\"backdrop-blur-sm\\" closable=\\"false\\">
  <div class=\\"flex flex-col px-8 py-8 gap-6 rounded-2xl\\">
    <div class=\\"inline-flex flex-col gap-2\\"><label for=\\"username\\" class=\\"text-primary-50 font-semibold\\">Username</label>
      <prime-input-stub id=\\"username\\"></prime-input-stub>
    </div>
    <div class=\\"inline-flex flex-col gap-2\\"><label for=\\"password\\" class=\\"text-primary-50 font-semibold\\">Password</label>
      <prime-input-stub id=\\"password\\" type=\\"password\\"></prime-input-stub>
    </div>
    <div class=\\"flex items-center gap-4\\">
      <prime-button-stub label=\\"Sign-In\\" variant=\\"text\\" class=\\"!p-4 w-full !text-primary-50 !border !border-white/30 hover:!bg-white/10\\"></prime-button-stub>
    </div>
  </div>
</div>"
`;

exports[`LoginModal.vue > matches snapshot 1`] = `
"<div class=\\"prime-dialog-stub\\" visible=\\"true\\" pt:root:class=\\"!border-0 !bg-transparent\\" pt:mask:class=\\"backdrop-blur-sm\\" closable=\\"false\\">
  <div class=\\"flex flex-col px-8 py-8 gap-6 rounded-2xl\\">
    <div class=\\"inline-flex flex-col gap-2\\"><label for=\\"username\\" class=\\"text-primary-50 font-semibold\\">Username</label>
      <prime-input-stub id=\\"username\\"></prime-input-stub>
    </div>
    <div class=\\"inline-flex flex-col gap-2\\"><label for=\\"password\\" class=\\"text-primary-50 font-semibold\\">Password</label>
      <prime-input-stub id=\\"password\\" type=\\"password\\"></prime-input-stub>
    </div>
    <div class=\\"flex items-center gap-4\\">
      <prime-button-stub label=\\"Sign-In\\" variant=\\"text\\" class=\\"!p-4 w-full !text-primary-50 !border !border-white/30 hover:!bg-white/10\\"></prime-button-stub>
    </div>
  </div>
</div>"
`;
